src/test/java/teammates/sqlui/webapi/RegenerateInstructorKeyActionTest.java

```diff
@@ -0,0 +1,221 @@
+                instructor.getCourseId(),
+                instructor.getEmail(),
+                EmailType.INSTRUCTOR_COURSE_LINKS_REGENERATED
+        )).thenReturn(mockEmail);
+        mockEmailSender.setShouldFail(false);
+    }
+
+    @Test
+    void testExecute_successfulRegenerationWithEmailSent_success()
+            throws EntityDoesNotExistException, InstructorUpdateException {
+        when(mockLogic.regenerateInstructorRegistrationKey(instructor.getCourseId(), instructor.getEmail()))
+                .thenReturn(instructor);
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, instructor.getCourseId(),
+                Const.ParamsNames.INSTRUCTOR_EMAIL, instructor.getEmail(),
+        };
+
+        RegenerateInstructorKeyAction action = getAction(params);
+        RegenerateKeyData actionOutput = (RegenerateKeyData) getJsonResult(action).getOutput();
+
+        verify(mockLogic, times(1)).regenerateInstructorRegistrationKey(instructor.getCourseId(), instructor.getEmail());
+        verify(mockSqlEmailGenerator, times(1)).generateFeedbackSessionSummaryOfCourse(
+                instructor.getCourseId(),
+                instructor.getEmail(),
+                EmailType.INSTRUCTOR_COURSE_LINKS_REGENERATED
+        );
+        verifyNumberOfEmailsSent(1);
+        verifyNoMoreInteractions(mockLogic, mockSqlEmailGenerator);
+        assertEquals(SUCCESSFUL_REGENERATION_WITH_EMAIL_SENT, actionOutput.getMessage());
+        assertNotNull(actionOutput.getNewRegistrationKey());
+    }
+
+    @Test
+    void testExecute_successfulRegenerationWithEmailFailed_success()
+            throws EntityDoesNotExistException, InstructorUpdateException {
+        when(mockLogic.regenerateInstructorRegistrationKey(instructor.getCourseId(), instructor.getEmail()))
+                .thenReturn(instructor);
+        mockEmailSender.setShouldFail(true);
+
+        String[] params = {
+                Const.ParamsNames.COURSE_ID, instructor.getCourseId(),
+                Const.ParamsNames.INSTRUCTOR_EMAIL, instructor.getEmail(),
+        };
+
+        RegenerateInstructorKeyAction action = getAction(params);
+        RegenerateKeyData actionOutput = (RegenerateKeyData) getJsonResult(action).getOutput();
+
+        verify(mockLogic, times(1)).regenerateInstructorRegistrationKey(instructor.getCourseId(), instructor.getEmail());
+        verify(mockSqlEmailGenerator, times(1)).generateFeedbackSessionSummaryOfCourse(
+                instructor.getCourseId(),
+                instructor.getEmail(),
+                EmailType.INSTRUCTOR_COURSE_LINKS_REGENERATED
+        );
+        verifyNoEmailsSent();
+        verifyNoMoreInteractions(mockLogic, mockSqlEmailGenerator);
+        assertEquals(SUCCESSFUL_REGENERATION_BUT_EMAIL_FAILED, actionOutput.getMessage());
+        assertNotNull(actionOutput.getNewRegistrationKey());
+    }
+
+    @Test
+    void testExecute_entityDoesNotExist_throwsEntityNotFoundException()
```

Comment:
nit: rename this to instructorDoesNotExist... to make it clear that it's referring to a non-existent instructor

Reply:
additionally, can we also copy over the test for non-existent course?